src/Courses/TopicSettings/ProblemSettings.tsx

```diff
@@ -49,12 +49,13 @@ export const ProblemSettings: React.FC<ProblemSettingsProps> = ({selected, setSe
 
     const topicForm = useForm<ProblemSettingsInputs>(formSettings);
 
-    const { handleSubmit, control, watch, reset, setError } = topicForm;
+    const { handleSubmit, control, watch, reset, setError, clearErrors } = topicForm;
     const { optional, smaEnabled, webworkQuestionPath } = watch();
 
     const additionalProblemPaths = watch('courseQuestionAssessmentInfo.additionalProblemPaths', [{path: ''}]);
     const [{ message: updateAlertMsg, severity: updateAlertType }, setUpdateAlert] = useMUIAlertState();
-    const [PGErrorsMsg, setPGErrorsAlert] = useState<string[]>([]);
+    const noPGErrorObject = useRef([]);
```

Comment:
What is the point of noPGErrorObject, is it just an optimization?

Reply:
Yes, it's just an optimization.

Creating a new array every time to null the errors in a setState causes unnecessary rerenders because the address changes.